islandora_solr_config/includes/atom_results.inc

```diff
@@ -0,0 +1,248 @@
+<?php
+
+/**
+ * @file
+ * Contains methods to search solr and display results.  depends on
+ * Apache_Solr_Php client.
+ */
+
+/**
+ * Extention of IslandoraSolrResults for templating purposes.
+ * This overrides the displayResults function to provide an alternate display
+ * type.
+ */
+
+class IslandoraSolrResultsAtom extends IslandoraSolrResults {
+
+  /**
+   * Outputs results basically in the normal way.
+   *
+   * Thumbnails pulled from the Fedora repository.
+   *
+   * @param object $islandora_solr_query
+   *   A solr query object.
+   *
+   * @return string
+   *   html output for the resultset. Note: we currently create this
+   *   output manually, should refactor to use drupal forms api.
+   */
+  public function printAtom($islandora_solr_query, $title = "Search Results") {
+
+    global $base_url;
+
+    drupal_add_http_header('Content-Type', 'application/atom+xml; charset=utf-8');
+    $islandora_solr_query->solrLimit = variable_get('islandora_solr_config_rss_limit', 50);
+    $islandora_solr_query->executeQuery();
+
+    // Get raw results.
+    $solr_result = $islandora_solr_query->islandoraSolrResult;
+
+    // All results.
+    $docs = $solr_result['response']['objects'];
+
+    // Loop over results.
+    $items = NULL;
+    foreach ($docs as $doc) {
+      // Turn arrays into strings.
+      foreach ($doc['solr_doc'] as $key => $value) {
+        if (is_array($value)) {
+          // Turn array into comma separated string and trim.
+          $doc[$key] = trim(implode(', ', $value));
+        }
+        else {
+          // Give it a trim.
+          $doc[$key] = trim($value);
+        }
+      }
+      // Get the variables for the <item> element.
+      $item = $this->rssItem($doc);
+
+      // Hook alter to change the rssItem before formatting.
+      drupal_alter('islandora_solr_search_rss_item', $item, $doc);
+
+      // Render rss item.
+      $rendered_item = format_rss_item($item['title'], $item['link'], $item['description'], $item['items']);
+      $open_tag = "<item>";
+      $close_tag = "</item>";
+      $rendered_item = str_replace($open_tag, "<entry>", $rendered_item);
+      $rendered_item = str_replace($close_tag, "</entry>", $rendered_item);
```

Comment:
I don't like these `str_replace` bits. I think it would be better to create a `format_atom_item` function in this class and output what we want, then to `str_replace` generated RSS.

Reply:
I was trying that but not having much success. `format_rss_item` and `format_rss_channel` are Drupal functions, so I tried creating my own versions using the code at https://api.drupal.org/api/drupal/includes%21common.inc/function/format_rss_item/7.x

But even if I just copy the code and change the function name, I get a 500 error. No syntax errors. Any idea what's happening there?